samples/bluetooth/central_hids/src/main.c

```diff
@@ -247,6 +331,139 @@ static const struct bt_gatt_hids_c_init_params hids_c_init_params = {
 	.pm_update_cb  = hids_c_pm_update_cb
 };
 
+
+static void button_bootmode(void)
+{
+	if (!bt_gatt_hids_ready_check(&hids_c)) {
+		printk("HID device not ready\n");
+		return;
+	}
+	int err;
+	enum bt_gatt_hids_c_pm pm = bt_gatt_hids_c_pm_get(&hids_c);
+
+	printk("Setting protocol mode: %s\n",
+	       (pm == BT_GATT_HIDS_C_PM_BOOT) ?
+	       "BOOT" : "REPORT");
+	err = bt_gatt_hids_c_pm_write(&hids_c,
+				      (pm == BT_GATT_HIDS_C_PM_BOOT) ?
+				      BT_GATT_HIDS_C_PM_REPORT :
+				      BT_GATT_HIDS_C_PM_BOOT);
+	if (err) {
+		printk("Cannot change protocol mode (err %d)\n", err);
+	}
+}
+
+
+static void button_capslock(void)
+{
+	if (!bt_gatt_hids_ready_check(&hids_c)) {
+		printk("HID device not ready\n");
+		return;
+	}
+	if (!hids_c.rep_boot.kbd_out) {
+		printk("HID device does not have Keyboard OUT report\n");
+		return;
+	}
+	if (bt_gatt_hids_c_pm_get(&hids_c) != BT_GATT_HIDS_C_PM_BOOT) {
+		printk("This function works only in BOOT Report mode\n");
+		return;
+	}
+	int err;
```

Comment:
Nitpicking: if you could move the declarations at the top of the function it'd be better, at least boilerplate variable like `err`.

Reply:
Done